### PR TITLE
Api data

### PR DESCRIPTION
**Pull request by dkwon1223**
# Summary 

This update addresses the following user story requirements:
- When the server returns a 500 error, the user will see proper error handling
  - Error handling was implemented by throwing new errors within fetch calls through conditional logic in responses. 
  - `useState` was implemented to set the error message and error status as props to pass down to a new error handling component which conditionally renders errors messages to users
- No other user-facing features required
  - yay
- We’re mostly refactoring in this iteration!
  - refactoring: 
    - fetching data
      - 'useEffect' for all movies
    - error handling
    - detailed movie styling
      - this includes a new npm package react-stars for dynamic UI component for movie rating

# To do 
- Upcoming: Erin's amazing updates with search functionality and PropTypes
- Refactoring with Router
- TDD with Cypress

### CODE REVIEW

File: package.json
Comment (kelleyej) [inline package.json:1]: Awesome package addition with the stars rating! Greatly improves the UI!

File: src/components/App.js
Comment (kelleyej) [inline src/components/App.js:1]: Nice error handling implementation for both 400 and 500 level status codes!

File: src/components/App.js [hunk 1]
```diff
@@ -2,28 +2,65 @@ import '../styles/App.scss';
...
+            .catch(error => {
+              setError(error.message);
+            })
   }
 
   return (
     <main className='App'>
       <Navbar />
       {details ? <MovieDetails movie={movie} returnToHome={returnToHome}/> : <MoviesArea getMovieDetails={getMovieDetails} movies={movies} />}
+      {error !== null && <ErrorMessage error={error} errorStatus={errorStatus} />}
```
Comment (kelleyej) [inline src/components/App.js:63]: Possibly refactor to say error && <ErrorMessage ... > ?

File: src/components/ErrorMessage.js
Comment (kelleyej) [inline src/components/ErrorMessage.js:1]: Loved that you added an error component! I feel like that improves the user experience.

File: src/components/Movie.js
Comment (kelleyej) [inline src/components/Movie.js:1]: Looks great consistently refactoring across files for the export default syntax.

File: src/styles/ErrorMessage.scss
Comment (kelleyej) [inline src/styles/ErrorMessage.scss:1]: Error message looks great and really explains to the user exactly what is going wrong.

**Review by kelleyej — COMMENTED** [review]
Cool feature adding the star rating package! Great job with use of useEffect for fetching data, implementing error handling, and improving user experience. The UI looks awesome!

File: src/components/App.js (hunk 1 above)
Comment (dkwon1223) [reply src/components/App.js:63]: Great catch!

File: src/components/App.js [hunk 2]
```diff
@@ -2,28 +2,65 @@ import '../styles/App.scss';
...
   const [details, setDetails] = useState(false);
+  const [error, setError] = useState(null);
+  const [errorStatus, setErrorStatus] = useState(null)
  
+  async function getAllMovies() {
+    await fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
+            .then(response => {
+              if(!response.ok) {
+                setErrorStatus(response.status);
+                throw new Error("Unable to retrieve movies. Try again later.");
```
Comment (nicktu12) [inline src/components/App.js:20]: How does this error get shown to the user? Its common to have `error` in state, and then based on what that variable is - do some conditional rendering to should the user helpful messaging. Thrown errors arent the best UX imo.

File: src/components/App.js (hunk 1 above)
Comment (nicktu12) [reply src/components/App.js:63]: YAS!

File: src/components/ErrorMessage.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import '../styles/ErrorMessage.scss';
+
+export default function ErrorMessage({ error, errorStatus }) {
+  return (
+    <div className='error-container'>
+      <h1 className='error-message'>{error}<br/>STATUS CODE {errorStatus}</h1>
+      <button onClick={() => {window.location.reload()}}>Back to Homepage</button>
```
Comment (nicktu12) [inline src/components/ErrorMessage.js:8]: Router will help with this :)

File: src/components/MovieDetails.js [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
 
-function MovieDetails({ returnToHome, movie }){
+export default function MovieDetails({ returnToHome, movie }){
     return (
         <section className="detail-container">
-            <h1>{movie.title}</h1>
-            <img src={movie.backdrop_path}/>
-            <button onClick={returnToHome}>Back to Homepage</button> 
+            <article>
+                <p>{movie.runtime} min | {movie.release_date.slice(0,4)}</p>
```
Comment (nicktu12) [inline src/components/MovieDetails.js:8]: Day.js, moment.js are helpful libraries for dealing with dates!!!

**Pull request merged**